homeassistant/components/media_player/bluesound.py

```diff
@@ -0,0 +1,812 @@
+        self._services_items = []
+        self._preset_items = []
+        self._sync_status = {}
+        self._status = None
+        self._last_status_update = None
+        self._is_online = False
+        self._retry_remove = None
+        self._lastvol = None
+        self._init_callback = init_callback
+        if self._port is None:
+            self._port = DEFAULT_PORT
+
+# Internal methods
+    @staticmethod
+    def _try_get_index(string, seach_string):
+        try:
+            return string.index(seach_string)
+        except ValueError:
+            return -1
+
+    @asyncio.coroutine
+    def _internal_update_sync_status(self, on_updated_cb=None,
+                                     raise_timeout=False):
+        resp = None
+        try:
+            resp = yield from self.send_bluesound_command(
+                'SyncStatus',
+                raise_timeout, raise_timeout)
+        except:
+            raise
+
+        if not resp:
+            return None
+        self._sync_status = resp['SyncStatus'].copy()
+
+        if not self._name:
+            self._name = self._sync_status.get('@name', self.host)
+        if not self._brand:
+            self._brand = self._sync_status.get('@brand', self.host)
+        if not self._model:
+            self._model = self._sync_status.get('@model', self.host)
+        if not self._icon:
+            self._icon = self._sync_status.get('@icon', self.host)
+        if not self._model_name:
+            self._model_name = self._sync_status.get('@modelName', self.host)
+
+        if on_updated_cb:
+            on_updated_cb()
+        return True
+# END Internal methods
+
+# Poll functionality
+    @asyncio.coroutine
+    def _start_poll_command(self):
+        """"Loop which polls the status of the player."""
+        try:
+            while True:
+                yield from self.async_update_status()
+
+        except (asyncio.TimeoutError, ClientError):
+            _LOGGER.info("Bluesound node %s is offline, retrying later",
+                         self._name)
+            yield from asyncio.sleep(NODE_OFFLINE_CHECK_TIMEOUT,
+                                     loop=self._hass.loop)
+            self.start_polling()
+
+        except CancelledError:
+            _LOGGER.debug("Stopping bluesound polling of node %s", self._name)
+        except:
+            _LOGGER.exception("Unexpected error in %s", self._name)
+            raise
+
+    def start_polling(self):
+        """Start the polling task."""
+        self._polling_task = self._hass.async_add_job(
+            self._start_poll_command())
+
+    def stop_polling(self):
+        """Stop the polling task."""
+        self._polling_task.cancel()
+# END Poll functionality
+
+# Initiator
+    @asyncio.coroutine
+    def async_init(self, what=None):
+        """Initiate the player async."""
+        if what is None:
+            # if it's the first request.
+            # let's make the request in a new thread
```

Comment:
It's because we track all tasks during startup and will wait for them to finish.

If you want to enqueue a task by avoiding this, do not use a time tracking thing. That's a huge hack. Instead, listen for the `EVENT_HOMEASSISTANT_START` event to start the initialization. If Home Assistant is already running you can start init right away.